k9mail/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -175,6 +185,27 @@ public void processMessageToForward(MessageViewInfo messageViewInfo) {
         }
     }
 
+    public void processMessageToForwardAsAttachment(MessageViewInfo messageViewInfo) throws IOException, MessagingException {
+        boolean isMissingParts = !allPartsAvailable(messageViewInfo, false);
+        if (isMissingParts) {
+            attachmentMvpView.showMissingAttachmentsPartialMessageForwardWarning();
+        } else {
+            File tempFile = File.createTempFile("pre", ".tmp");
+            tempFile.deleteOnExit();
+            FileOutputStream fileOutputStream = new FileOutputStream(tempFile);
+            messageViewInfo.message.writeTo(fileOutputStream);
```

Comment:
this should not be done on the UI thread but rather in a loader. check out `AttachmentContentLoader`